python_package/madflow/custom_op_generator.py

```diff
@@ -0,0 +1,165 @@
+import subprocess
+
+
+import madflow.wavefunctions_flow
+from madflow.makefile_template import write_makefile
+
+from madflow.op_write_templates import *
+from madflow.op_syntax import *
+from madflow.op_read import *
+from madflow.op_generation import *
+
+
+folder_name = "prov/"
+
+temp = ""
+devices = ["cpu", "gpu"]
+
+
+def translate(destination):
+    """Translates Python code into a C++/CUDA Custom Operator
+    destination: directory of madflow output"""
+
+    if destination[-1] != "/":  # Avoid weird behaviours if destination does not end with '/'
+        destination += "/"
+    file_sources = [madflow.wavefunctions_flow.__file__]  # path to wavefunctions_flow.py
+
+    # Create the directory for the Op source code and create the makefile
+
+    subprocess.check_output(["/bin/sh", "-c", "mkdir -p " + destination + "gpu/"])
+    write_makefile(destination)
+
+    # Generate sign functions
+    auxiliary_functions = []
+    function_list_ = []
+    auxiliary_functions, function_list_ = generate_auxiliary_functions(
+        auxiliary_functions, function_list_
+    )
+
+    # Read wavefunctions_flow.py
+    for file_source in file_sources:
+        signatures_ = []
+        signature_variables_ = []
+
+        signatures_, signature_variables_ = read_signatures(
+            signatures_, signature_variables_, file_source
+        )
+
+        signature_variables_ = convert_signatures(signatures_, signature_variables_)
+
+        function_list_ = read_file_from_source(
+            function_list_, file_source, signatures_, signature_variables_
+        )
+
+    # Find all generated matrix_1_xxxxx.py (one for each subprocess)
+    files_list = (
+        subprocess.check_output(["/bin/sh", "-c", "ls " + destination + " | grep matrix_1_"])
+        .decode("utf-8")
+        .split("\n")[:-1]
+    )
+
+    for _file_ in files_list:
+
+        constants = []  # globalConstants
+
+        for e in globalConstants:
+            constants.append(e)
+
+        process_name = re.sub("matrix_1_", "", _file_)
+        process_name = re.sub("\.py", "", process_name)
+
+        matrix_source = destination + "matrix_1_" + process_name + ".py"
+        process_source = destination + "aloha_1_" + process_name + ".py"
+
+        _file_ = process_source
```

Comment:
I would not change the value of `_file_` here, just use `process_source` when you need it.

Reply:
I commented out line 74 (to be removed in the next commit) and used `process_source`.